pagopa/io-app
# chore: [IOBP-533] Navigation to payment confirm screen if method with only one PSP

## Short description
This PR implements the user experience enhancement while a specific payment method selected during the payment process has only one PSP, it automatically selects it and goes forward to the Payment confirmation screen without choosing the PSP since it's the only one in the list.

## List of changes proposed in this pull request
- Moved the `calculateFee` API request into `WalletPaymentPickMethodScreen` while the user taps on the "Continue" button;
- Added an additional standard redux action `walletPaymentResetPspList` to reset the PSP list when entering into the screen;

## How to test
- Start a new payment flow into the playground;
- Try to return only one PSP from the `bundle` list located on the `io-dev-app-server`, more specifically into the `src/features/wallet/payloads/transactions.ts` file
- Then try to continue selecting a payment method, you should be able to go forward directly to the confirmation screen.

## Preview

https://github.com/pagopa/io-app/assets/34343582/6f5e7610-0982-4305-87e3-cf9fe3e79129
